src/main/java/spot/Parser.java

```diff
@@ -0,0 +1,134 @@
+package spot;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
+
+import spot.command.AddDeadlineCommand;
+import spot.command.AddEventCommand;
+import spot.command.AddToDoCommand;
+import spot.command.Command;
+import spot.command.DeleteCommand;
+import spot.command.ExitCommand;
+import spot.command.FindCommand;
+import spot.command.ListCommand;
+import spot.command.ListTasksOnCommand;
+import spot.command.MarkCommand;
+import spot.command.UnmarkCommand;
+import spot.exception.SpotException;
+
+
+/**
+ * Represents a parser to parse commands and dates.
+ */
+public class Parser {
+
+    /**
+     * Returns command corresponding to the specified input string.
+     *
+     * @param input Input string to be parsed.
+     * @return Resulting command.
+     * @throws SpotException  If input string is invalid.
+     */
+    public static Command parseCommand(String input) throws SpotException {
```

Comment:
parseCommand is quite long and does a lot of things, might be beneficial to break it down or add in-line comments to add clarity.